weblate/templates/translation.html

```diff
@@ -234,6 +238,33 @@ <h4 class="panel-title">
 <p>{% trans "Automatic translation takes existing translations in this project and applies them to the current component. It can be used to push translations to a different branch, to fix inconsistent translations or to translate a new component using translation memory." %}</p>
 <p>{% trans "Automatic translation via machine translation uses active machine translation engines to get the best possible translations and applies them in this project." %}</p>
 {% crispy autoform %}
+<script src="/static/vendor/bootstrap/js/tom-select.base.js"></script>
+<script>
+new TomSelect('#id_component', {
+    valueField: 'id',
+    labelField: 'name',
+    searchField: 'name',
+    maxItems: 1,
+    closeAfterSelect: true,
+    allowEmptyOption: true,
+    load: function(query, callback) {
+        let url = '/api/components/shared/?search=' + encodeURIComponent(query);
+        fetch(url)
+            .then(response => response.json())
+            .then(json => {
+                callback(json.results);
+            })
+            .catch(() => {
+                callback();
+            });
+    },
+    render: {
+        option: function(item, escape) {
+             return `<div><h4 class="list-group-item-heading">${ escape(item.name) }</h4><div class="list-group-item-text text-primary">${ escape(item.project.name) }</div></div>`;
+        },
+    },
+});
```

Comment:
Please avoid inline scripts, these are blockec by Content-Security-Policy.

Reply:
Ok, I'll move it in a static file. Maybe there is already a file into which I could inject that with some refactor so it's only valid for this form?

> Ideally all needed parts would be rendered by the form itself.

I'm not sure I follow you. Does it mean I should find a way to make the class rendering the added part in the template file ?

Reply:
You can add needed CSS and scripts to [weblate/templates/addons/addon_detail.html](https://github.com/WeblateOrg/weblate/blob/main/weblate/templates/addons/addon_detail.html). It's acceptable for this single use case, but once there will be more features like this, it will need refactoring to avoid loading too much of unused code.

Another option might be using ESM to load the module, this has been outlined in https://github.com/WeblateOrg/weblate/pull/7586, but it has its challenges as well. But maybe introducing it for new code might be easier than converting existing one.